translate traits

yodalee opened the pull request:
translate trait

yodalee committed:
translate traits

yodalee commented on src/traits.md:
```diff
@@ -404,14 +384,13 @@ fn inverse<T>() -> T
-`ConvertTo<T>`. Rather than defining what `i32` is (since that's obvious), the
-`where` clause here constrains `T`.
+這展示 `where` 額外的特性，它們除了型別參數 `T`，亦可限制型別（範例中的 `i32`)
+在範例中，`i32` 必須實作 `ConvertTo<T>`，這裡的 `where` 限制 `T`而不是定義 `i32`是什麼
```
This section, I'm not sure how to translate it.

askeing replied:
I am also confused at `Rather than defining what i32 is (since that's obvious), the where clause here constrains T.`

yodalee replied:
我覺得我先保留原文好了

askeing commented on src/traits.md:
```diff
@@ -1,10 +1,9 @@
 
-A trait is a language feature that tells the Rust compiler about
-functionality a type must provide.
+trait 為Rust 語言特徵，告知Rust compiler 一個型別必須滿足的功能性
```
英文前後請留一個空白 `為 Rust 語言特徵`，句尾請加上全形句號 `。`
ref: [中文文案排版指南](https://github.com/sparanoid/chinese-copywriting-guidelines)

askeing commented on src/traits.md:
```diff
@@ -1,10 +1,9 @@
 
-Recall the `impl` keyword, used to call a function with [method
-syntax][methodsyntax]:
+記得提過的 `impl` 關鍵字，用在使用[method syntax][methodsyntax]
```
link 的前後請加空白，另外句尾也需要句號。

askeing commented on src/traits.md:
使用全形括號，以及加上全形逗號，建議可以改為這樣：

``` markdown
Traits 保證一個型別應有的行為，因此非常有用。
泛型函式能利用 trait 作為 [限制][bounds]（bound），用來限制它接受的型別。
```

askeing commented on src/traits.md:
````diff
@@ -253,15 +243,15 @@ let result = f.write(buf);
 ```
 
-Here’s the error:
+以下為編譯錯諤：
````
typo here

yodalee committed:
Fix some small problem: space, punctuate:

add space before and after english
use full-sized punctuate
add English paragraph that don't know how to translate

yodalee commented:
mostly fixed

askeing commented:
@yodalee thx :)
